sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1076,6 +1078,14 @@ def _decorated(*args, **kwargs):
                 # only proxy public callables
                 setattr(self, attr_to_proxy, _proxy(attr_to_proxy))
 
+        # Special handling for the mlpipeline-ui-metadata and mlpipeline-metrics outputs that should always be saved as artifacts
+        # TODO: Remove when outputs are always saved as artifacts
+        for output_name, path in dict(file_outputs).items():
+            normalized_output_name = re.sub('[^a-zA-Z0-9]', '-', output_name.lower())
+            if normalized_output_name in ['mlpipeline-ui-metadata', 'mlpipeline-metrics']:
+                output_artifact_paths[normalized_output_name] = path
+                del file_outputs[output_name]
```

Comment:
My thought is that these lines are a bit "dirty", but I get it has to deal with the existing implementation of `component`s which does not support artifacts.
Please see my review comment where I attempt to explain myself.

Reply:
This part is a temporary workaround.